projects/plugins/crm/includes/ZeroBSCRM.Core.Page.Controller.php

```diff
@@ -66,51 +66,36 @@ function zeroBSCRM_pages_admin_addedit() {
 // } This is a slow general move to new UI (and new DB) and moving away from the custom post add / edit links
 function zeroBSCRM_pages_admin_addedit_page( $type = 'contact', $action = 'new', $id = -1 ) {
 
-	// } learn script (for this page)
-
-	?>
-
-	<script type="text/javascript">
-
-		jQuery(function($){
-
-			jQuery('.learn')
-			.popup({
-				inline: false,
-				on:'click',
-				lastResort: 'bottom right',
-			});
-
-		});
-	</script>
-
-
-
+	global $zbs;
+	$obj_type_id = $zbs->DAL->objTypeID( $type ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 
-	<?php
-	switch ( $type ) {
+	switch ( $obj_type_id ) {
 
-		case 'contact':
+		case ZBS_TYPE_CONTACT:
 			// pass them this way..
 			zeroBSCRM_pages_admin_addedit_page_contact( $id, $action );
 			break;
 
-		case 'company':
+		case ZBS_TYPE_COMPANY:
 			zeroBSCRM_pages_admin_addedit_page_company( $id, $action );
 			break;
 
-		case 'segment':
+		case ZBS_TYPE_SEGMENT:
 			zeroBSCRM_pages_admin_addedit_page_segment( $id, $action );
 			break;
 
 		// DAL3.0 + the rest can be fired via zeroBSCRM_pages_admin_addedit_page_generic
-		case 'quote':
-		case 'invoice':
-		case 'transaction':
-		case 'event':
-		case 'form':
-		case 'quotetemplate':
-			zeroBSCRM_pages_admin_addedit_page_generic( $id, $action );
+		case ZBS_TYPE_QUOTE:
+		case ZBS_TYPE_INVOICE:
+		case ZBS_TYPE_TRANSACTION:
+		case ZBS_TYPE_EVENT:
+		case ZBS_TYPE_FORM:
+		case ZBS_TYPE_QUOTETEMPLATE:
+			if ( zeroBSCRM_permsObjType( $obj_type_id ) ) {
+				zeroBSCRM_pages_admin_addedit_page_generic( $id, $action );
+			} else {
+				echo esc_html( sprintf( __( 'You do not have permission to edit this %s.', 'zero-bs-crm' ), $zbs->DAL->typeStr( $obj_type_id ) ) ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase,WordPress.WP.I18n.MissingTranslatorsComment
```

Comment:
Looks like there are some spacing issues here on smaller screens:
<img width="369" alt="Screenshot 2023-06-04 at 16 02 02" src="https://github.com/Automattic/jetpack/assets/16754605/e967102c-eec8-4896-9b71-86cf2bf25af0">
<img width="524" alt="Screenshot 2023-06-04 at 16 02 16" src="https://github.com/Automattic/jetpack/assets/16754605/6721f1e4-18ec-4a95-82a0-e21af8a95932">

Reply:
The more general issue with margin/padding is already addressed in #31157, butI've pushed the error message slightly to the right in 35ffd86734 so it lines up with the page title.

Reply:
Looks good!